### ISSUE TITLE
enhancement: contact form field names from URL without postID

### ISSUE THREAD

**Issue by airdrummer** [Automattic/jetpack#10178]
currently, field names specified in URL params must include the postID. while not an onerous requirement, it is beyond the capabilities of many wordpress users to inspect their forms' source code; elimiating this hurdle would be helpful to them.

**Comment by kraftbj** [comment]
Thanks for the suggestion! I think a more straightforward approach would be to expose the field id somewhere instead of trying to change them to be something predictable. The URL of a page could change, for example.

**Comment by airdrummer** [comment]
why can't the plugin simply prepend "g".$postID? or isn't the pageID available to the plugin when the URL is parsed?

**Comment by kraftbj** [comment]
True, that's a good point. The actual field ID might not be what's expected as folks add more form fields, though, so the original difficulty of discoverability is still present. If Jetpack prepended, someone would need to know the non-prepended field ID or know to not include it in the args if discovering it via the front-end.

**Comment by airdrummer** [comment]
>  the original difficulty of discoverability
sorry, miscomm here: i'm trying to prefill fields on a form i created, so i already know the field names;-)

**Comment by kraftbj** [comment]
There is still a discoverability issue. The field label of "Website's Name! 😱 Put it here" (or using other characters in those character planes) yields an id that isn't always obvious. Thanks for your feedback and for suggesting enhancements for Jetpack!

**Comment by airdrummer** [comment]
sorry, i don't understand why your code can't simply use the postID of the post containing ur form:
`	$post_id = $wp_query->post->ID;
`
